src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -145,6 +183,34 @@ public bool Equals(ArraySegment<T> obj)
             return obj._array == _array && obj._offset == _offset && obj._count == _count;
         }
 
+        public ArraySegment<T> Slice(int index)
+        {
+            ThrowInvalidOperationIfDefault();
+            // Note: `index` is allowed to be negative so the start of this ArraySegment can be moved backwards.
```

Comment:
@jkotas @KrzysztofCwalina how well do negative bounds work for our optimized bounds checking pattern? 

```C#
if ((uint)index >= (uint)_count) ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
```

I think we want to make sure indexing is blazing fast, so I don't think we want to allow negative indices as it would cause additional branches. In slicing it might be OK performance wise, but would feel inconsistent if I can't do it for regular indexing.

It's also a pattern that doesn't exist in the BCL. I don't think we should introduce it here.